e2e/testcafe-devextreme/tests/editors/checkBox/common.ts

```diff
@@ -47,3 +59,110 @@ fixture.disablePageReloads`CheckBox`
+
+  await t
+    .expect(compareResults.isValid())
+    .ok(compareResults.errorMessages());
+}).before(async () => {
+  for (const iconSize of [20, undefined, 25]) {
+    for (const text of [undefined, 'label', 'one two three']) {
+      for (const state of [
+        READONLY_STATE_CLASS,
+        DEFAULT_STATE_CLASS,
+        ACTIVE_STATE_CLASS,
+        HOVER_STATE_CLASS,
+        FOCUSED_STATE_CLASS,
+        DISABLED_STATE_CLASS,
+      ] as any[]
+      ) {
+        for (const value of valueModes) {
+          const id = `dx${new Guid()}`;
+          await appendElementTo('#container', 'div', id, {});
+
+          await createWidget('dxCheckBox', {
+            text,
+            value,
+            iconSize,
+          }, `#${id}`);
+          await setClassAttribute(Selector(`#${id}`), state);
+        }
+      }
+    }
+  }
+});
+
+test('Checkbox configuration, different Checkbox widths', async (t) => {
+  const { takeScreenshot, compareResults } = createScreenshotsComparer(t);
+
+  await testScreenshot(t, takeScreenshot, 'CheckBox configurations different widths.png', { element: '#container', shouldTestInCompact: true });
+
+  await testScreenshot(t, takeScreenshot, 'CheckBox configurations different widths.png', { element: '#container', theme: getDarkThemeName() });
+
+  await t
+    .expect(compareResults.isValid())
+    .ok(compareResults.errorMessages());
+}).before(async () => {
+  for (const width of [45, undefined, 65]) {
+    for (const text of [undefined, 'label', 'one two three']) {
+      for (const state of [
+        READONLY_STATE_CLASS,
+        DEFAULT_STATE_CLASS,
+        ACTIVE_STATE_CLASS,
+        HOVER_STATE_CLASS,
+        FOCUSED_STATE_CLASS,
+        DISABLED_STATE_CLASS,
+      ] as any[]
+      ) {
+        for (const value of valueModes) {
+          const id = `dx${new Guid()}`;
+          await appendElementTo('#container', 'div', id, {});
+
+          await createWidget('dxCheckBox', {
+            text,
+            value,
+            width,
+          }, `#${id}`);
+          await setClassAttribute(Selector(`#${id}`), state);
+        }
+      }
+    }
+  }
+});
+
+test('Checkbox configuration, different Checkbox orientation', async (t) => {
+  const { takeScreenshot, compareResults } = createScreenshotsComparer(t);
+
+  await testScreenshot(t, takeScreenshot, 'CheckBox configurations different rtlOrientation.png', { element: '#container', shouldTestInCompact: true });
+
+  await testScreenshot(t, takeScreenshot, 'CheckBox configurations different rtlOrientation.png', { element: '#container', theme: getDarkThemeName() });
+
+  await t
+    .expect(compareResults.isValid())
+    .ok(compareResults.errorMessages());
+}).before(async () => {
+  for (const rtlEnabled of [false, true]) {
+    for (const width of [undefined, 45, 65]) {
+      for (const text of [undefined, 'label', 'one two three']) {
+        for (const value of valueModes) {
```

Comment:
dont think testing 'valueModes' in RTL tests are useful,  lets get rid of this option in this test